tests/utils/datatype_tests.py

```diff
@@ -74,3 +74,53 @@ def test_tile_clean(self):
         string.clean(tile2, nodeid)
 
         self.assertIsNotNone(tile2.data[nodeid])
+
+
+class URLDataTypeTests(ArchesTestCase):
+    def test_validate(self):
```

Comment:
Nice work with the tests; they're clear and they fail on the target branch.

One thing that can improve the developer experience when tests fail is when all of the assertions have a chance to fail. For example here, when I check them on the target branch, the test stops at the first failed assertion.

If you find yourself doing something repetitive, like calling `url.validate()`, you can batch them up and call them under `with self.subTest()` so that the runner will actually assert everything instead of failing fast and exiting.

Here's an example, where the left side `input=` is an arbitrary label that I want to appear in my test output:

https://github.com/archesproject/arches/blob/e4145e4cd546210c1bde18a04634fb36ec8d548b/tests/utils/datatype_tests.py#L33-L42

If you play with that example and change the expected result, you'll get all 9 failures with the info about the specific way each one failed. Versus if you remove the subTest() statement, you only get 1 failure.

We can merge this work without adding subTest IMO, but I just wanted to mention it in case you find this useful when writing other unit tests.